src/main/java/com/puppycrawl/tools/checkstyle/filters/PatchXpathFilterElement.java

```diff
@@ -27,16 +28,28 @@
 import com.puppycrawl.tools.checkstyle.TreeWalkerAuditEvent;
 import com.puppycrawl.tools.checkstyle.TreeWalkerFilter;
 import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
 
 /**
  * This filter element is immutable and processes.
  */
 public class PatchXpathFilterElement implements TreeWalkerFilter {
+    /**
+     * List of TokenTypes that can be as ancestor tokenTypes of checks
+     * from checkNamesForContextStrategyByTokenOrAncestorSet.
+     */
+    private final List<Integer> ancestorTokenSet =
```

Comment:
This fix doesn't depend on FinalLocalVariable. Please move it to a new commit.
And explain the problem in issue description (https://github.com/checkstyle/patch-filters/issues/163).

Reply:
done

Reply:
you still have changes related to checkNamesForContextStrategyByTokenOrAncestorSet in a commit for FinalLocalVariableCheck. Please move all changes related to this property to this commit. The other commit will have only FinalLocalVariable tests.

Reply:
done